src/query/storage/block.go

```diff
@@ -113,28 +179,24 @@ func (m *multiSeriesBlockStepIter) Meta() block.Metadata {
 }
 
 func (m *multiSeriesBlockStepIter) Next() bool {
-	if len(m.block.seriesList) == 0 {
+	if len(m.values) == 0 {
 		return false
 	}
 
 	m.index++
-	return m.index < m.block.StepCount()
+	return m.index < len(m.values[0])
 }
 
-func (m *multiSeriesBlockStepIter) Current() (block.Step, error) {
-	values := make([]float64, len(m.block.seriesList))
-	seriesLen := m.block.seriesList[0].Len()
-	for i, s := range m.block.seriesList {
-		if m.index < seriesLen {
-			values[i] = s.Values().ValueAt(m.index)
-		} else {
-			values[i] = math.NaN()
-		}
+func (m *multiSeriesBlockStepIter) Current() (block.UnconsolidatedStep, error) {
+	values := make([]ts.Datapoints, len(m.values))
+	for i, series := range m.values {
+		values[i] = series[m.index]
```

Comment:
Doesn't this need to pad with NaNs after the end too?

Reply:
we have all series here ? They are padded with NaNs themselves